### ISSUE TITLE
Command to substitute variables from launch config should also get workspaceFolder as an argument

### ISSUE THREAD

**Issue by isidorn** [microsoft/vscode#158471]
Issue raised by @nturinski:

We register the command `pickFuncProcess` which is triggered by pressing the Debug Start (F5) button. When we get the `DebugConfiguration` as the passed in the argument, it comes in as so:

![MicrosoftTeams-image (1)](https://user-images.githubusercontent.com/1926584/185382014-a346e4a2-b58f-4ea0-91b3-62ad2bd298c3.png)

The launch.json has `"processId": "${command:azureFunctions.pickProcess}"` set which is what triggers our `pickFuncProcess` command. 

Comment from @isidorn: The problem is that the Azure team can not differentiate what exact configuration it is in a Multi Root Workspace. Can we also pass the workspace folder as the second argument to command (the first argument is the configuration)?

**Comment by weinand** [comment]
Yes, we could pass additional parameters to the command about to be executed.

Problem: the "configuration" parameter is not documented, and as such is not yet public API.
So, before we introduce additional parameters, we should start documenting them and find a place where this documentation lives (may be https://code.visualstudio.com/api/references/commands ).

And then we should use the regular "Extension API process" for adding a "workspaceFolder" parameter.

**Comment by roblourens** [comment]
Sorry, I'm missing something, what's the relationship between `pickFuncProcess` and `azureFunctions.pickProcess`? And which workspace folder - not all debug configs are linked to a workspace folder. So what do they do when there isn't one?

Another option would be writing a `_workspaceFolder` property on every debug configuration when it's loaded from a launch.json. Could be useful in multiple scenarios.

**Comment by isidorn** [comment]
`pickFuncProcess` is just the internal name of their method, it is the implementation detail of the `azureFunctions.pickProcess` command. 
When there is no `workspaceFolder` then we would pass `undefined`. I believe our other debug api has the same contract - [example](https://github.com/microsoft/vscode/blob/main/src/vscode-dts/vscode.d.ts#L14286). When there isn't one the extension would know the config does not belong to any folder and can treat that case as they want.

Writing `_workspaceFolder` on every debug configuration sounds a bit hidden / not discoverable to me. Though that approach also works if you and @weinand prefer it.

**Comment by weinand** [comment]
I prefer the "folder" argument over  a `_workspaceFolder` on every debug configuration .
